### PR TITLE
MC-1438 Integrate MapBrowser into SQL Browser

### PR DESCRIPTION
**Pull request by Sodik**

### CODE REVIEW

**Comment by promanenko** [comment]
please update screenshots with the new design without "show first/show last ..." radiobuttons

File: docs/modules/tools/pages/sql-browser.adoc [hunk 1]
```diff
@@ -47,6 +47,14 @@ When the query is executed, the results are displayed under *Query Results*:
...
 
+=== Map browser
+
+If you run *simple SELECT queries (without JOIN and AGGREGATION)* over IMap Mappings then you can get access to xref:data-structures:map.adoc#map-browser[query medata].
+image:ROOT:SQLBrowserQueryResultTabWithClosedMapBrowser.png[A list of query results with chevron buttons]
+
+By clicking on *chevron button* at the beginning of each SQL result table row, metadata view opens below a row
+image:ROOT:SQLBrowserQueryResultTabWithOpenMapBrowser.png[A list of query results with metada view]
+
 You can view results in two modes:
```
Comment (promanenko) [inline docs/modules/tools/pages/sql-browser.adoc:58]: please remove the information about modes or state that they are applicable only to pre-5.2 clusters

File: docs/modules/tools/pages/sql-browser.adoc [hunk 2]
```diff
@@ -47,6 +47,14 @@ When the query is executed, the results are displayed under *Query Results*:
 .Example query results
 image:ROOT:SQLBrowserQueryResultTab.png[A list of query results]
 
+=== Map browser
+
+If you run *simple SELECT queries (without JOIN and AGGREGATION)* over IMap Mappings then you can get access to xref:data-structures:map.adoc#map-browser[query medata].
```
Comment (promanenko) [inline docs/modules/tools/pages/sql-browser.adoc:52]: query medata -> entry metadata?

File: docs/modules/tools/pages/sql-browser.adoc [hunk 3]
```diff
@@ -47,6 +47,14 @@ When the query is executed, the results are displayed under *Query Results*:
...
+image:ROOT:SQLBrowserQueryResultTabWithClosedMapBrowser.png[A list of query results with chevron buttons]
+
+By clicking on *chevron button* at the beginning of each SQL result table row, metadata view opens below a row
```
Comment (promanenko) [inline docs/modules/tools/pages/sql-browser.adoc:55]: is it `chevron button` or `chevron sign`?

File: docs/modules/tools/pages/sql-browser.adoc [hunk 4]
```diff
@@ -47,6 +47,14 @@ When the query is executed, the results are displayed under *Query Results*:
...
+image:ROOT:SQLBrowserQueryResultTabWithOpenMapBrowser.png[A list of query results with metada view]
```
Comment (promanenko) [inline docs/modules/tools/pages/sql-browser.adoc:56]: metada -> metadata?

File: docs/modules/tools/pages/sql-browser.adoc (hunk 1 above)
Comment (Sodik) [reply docs/modules/tools/pages/sql-browser.adoc:58]: removed

File: docs/modules/tools/pages/sql-browser.adoc (hunk 2 above)
Comment (Sodik) [reply docs/modules/tools/pages/sql-browser.adoc:52]: changed

File: docs/modules/tools/pages/sql-browser.adoc (hunk 3 above)
Comment (Sodik) [reply docs/modules/tools/pages/sql-browser.adoc:55]: button with chevron icon

File: docs/modules/tools/pages/sql-browser.adoc (hunk 4 above)
Comment (Sodik) [reply docs/modules/tools/pages/sql-browser.adoc:56]: done

File: docs/modules/tools/pages/sql-browser.adoc [hunk 5]
```diff
@@ -47,15 +47,13 @@ When the query is executed, the results are displayed under *Query Results*:
...
-* *Stream last 1000 records*: Display only the last 1000 records returned by the streaming query. The most recent record is always on the top of the list.
-+
-NOTE: This mode is only for displaying the results of streaming SQL queries.
+=== Map browser
 
+If you run *simple SELECT queries (without JOIN and AGGREGATION)* over IMap Mappings then you can get access to xref:data-structures:map.adoc#map-browser[entry metadata].
+image:ROOT:SQLBrowserQueryResultTabWithClosedMapBrowser.png[A list of query results with buttons]
 
+By clicking on *button with chevron icon* at the beginning of each SQL result table row, metadata view opens below a row
+image:ROOT:SQLBrowserQueryResultTabWithOpenMapBrowser.png[A list of query results with metadata view]
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:56]: This image displays a -1 for *Hits*. Is this expected?

File: docs/modules/tools/pages/sql-browser.adoc [hunk 6]
```diff
@@ -47,15 +47,13 @@ When the query is executed, the results are displayed under *Query Results*:
...
 image:ROOT:SQLBrowserQueryResultTab.png[A list of query results]
 
-You can view results in two modes:
-
-* *Show first 1000 records*: Display only the first 1000 records that are returned by the query.
-
-* *Stream last 1000 records*: Display only the last 1000 records returned by the streaming query. The most recent record is always on the top of the list.
-+
-NOTE: This mode is only for displaying the results of streaming SQL queries.
+=== Map browser
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:50]:
```suggestion
=== Viewing Map Metadata
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 7]
```diff
@@ -47,15 +47,13 @@ When the query is executed, the results are displayed under *Query Results*:
...
 
+If you run *simple SELECT queries (without JOIN and AGGREGATION)* over IMap Mappings then you can get access to xref:data-structures:map.adoc#map-browser[entry metadata].
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:52]:
```suggestion
If you run simple `SELECT` queries without joins or aggregations over maps, you can view xref:data-structures:map.adoc#map-browser[metadata] for each entry that is returned by the query.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 8]
```diff
@@ -47,15 +47,13 @@ When the query is executed, the results are displayed under *Query Results*:
...
+image:ROOT:SQLBrowserQueryResultTabWithClosedMapBrowser.png[A list of query results with buttons]
 
+By clicking on *button with chevron icon* at the beginning of each SQL result table row, metadata view opens below a row
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:55]:
```suggestion
To view an entry's metadata, click the chevron (*⌄*) button next to a map entry in the query results.
```

**Review by JakeSCahill — CHANGES_REQUESTED** [review]

File: docs/modules/tools/pages/sql-browser.adoc (hunk 5 above)
Comment (promanenko) [reply docs/modules/tools/pages/sql-browser.adoc:56]: it is a known bug on the member side, to be fixed before 5.2 release https://hazelcast.atlassian.net/browse/HZ-1313
@Sodik could you attach a screenshot with non-negative Hits value? it should work fine if there is a single member

**Review by promanenko — APPROVED** [review]

**Comment by JakeSCahill** [comment]
We should backport this to the `v/5.2-BETA-1` branch

**Pull request merged**